invenio/modules/search/api.py

```diff
@@ -59,6 +59,13 @@ def query(self):
     def search(self, user_info=None, collection=None):
         """Search records."""
         user_info = user_info or current_user
+        # Enhance query first
+        self.enhanced_query = self.query
+        for enhancer in cfg['SEARCH_QUERY_ENHANCERS']:
```

Comment:
you can cache the list of query enhances, but I would not add anything to `self.enhanced_query` because it depends on arguments the `search` method gets.